packages/realm-web/src/OAuth2Helper.ts

```diff
@@ -206,10 +212,22 @@ export class OAuth2Helper {
                     }
                 }
             };
+
             // Add a listener to the state storage, awaiting an update to the secret
             stateStorage.addListener(handleStorageUpdate);
             // Open up a window
             redirectWindow = this.openWindow(url);
+            // No using a const, because we need the two listeners to reference each other when removing the other.
+            windowClosedInterval = setInterval(() => {
```

Comment:
I'm using `setInterval` to poll the `window.closed` property, because apparently adding an event listener on the "close" event of on the newly created window is a cross-origin violation.